Framework/PythonInterface/mantid/simpleapi.py

```diff
@@ -929,10 +930,15 @@ def set_properties(alg_object, *args, **kwargs):
     def do_set_property(name, new_value):
         if new_value is None:
             return
-        if isinstance(new_value, _kernel.DataItem) and new_value.name():
-            alg_object.setPropertyValue(key, new_value.name())
-        else:
-            alg_object.setProperty(key, new_value)
+        try:
+            if isinstance(new_value, _kernel.DataItem) and new_value.name():
+                alg_object.setPropertyValue(key, new_value.name())
+            else:
+                alg_object.setProperty(key, new_value)
+        except RuntimeError as e:
+            msg = 'Problem setting "{}" in {}-v{}: {}'.format(name, alg_object.name(), alg_object.version(),
+                                                              str(e))
+            raise e.__class__(msg) from e
```

Comment:
This is weird coding. `e.__class__` is always `RuntimeError` here, so why not just raise that? You don't process any other error types

Reply:
I could have sworn that you get the correct subclass (e.g. `ValueError`) rather than the superclass (i.e. `RuntimeError`). I can switch to the the `type(e)(msg) from e` syntax that you suggested in slack if that is clearer.

Reply:
Your code works. I'm surprised that `ValueError` and `TypeError` are both converted to `RuntimeError`.

Reply:
You catch the superclass, but the object is of its own correct (subclass) type